homeassistant/components/deconz/climate.py

```diff
@@ -231,16 +244,23 @@ async def async_set_preset_mode(self, preset_mode: str) -> None:
     @property
     def current_temperature(self) -> float:
         """Return the current temperature."""
+        assert isinstance(self._device.temperature, float)
```

Comment:
Ok, but this goes beyond the PR description.

Was this line intended for the other PR?  (Could change behaviour)

Reply:
This is needed because the library class is assumed to be Any since I haven't added py.typed file to the library yet, because that would be a much bigger rewrite of the whole integration

Reply:
Agreed on the first point, but I think asserts are only disabled if you call python with -O (or equivalent), otherwise they generate an AssertionError.

Reply:
I'd suggest to use either `cast(float, self._device.temperature)` (from typing import cast) 
or just `# type: ignore[no-any-return]` which I find has better readability.
When you add a py.typed to the lib mypy will remind you that these aren't needed anymore.

Reply:
Thank you, I will investigate these suggestions.

Reply:
@farmio I've updated the implementation based on your suggestions.